runtime/src/mimalloc/c/include/mimalloc-internal.h

```diff
@@ -457,14 +457,27 @@ static inline uintptr_t _mi_thread_id(void) mi_attr_noexcept {
 }
 #elif (defined(__GNUC__) || defined(__clang__)) && \
       (defined(__x86_64__) || defined(__i386__) || defined(__arm__) || defined(__aarch64__))
+#if defined(KONAN_MI_MALLOC)
+  #include <pthread.h>
+  pthread_t pthread_self(void);
+#endif
 // TLS register on x86 is in the FS or GS register
 // see: https://akkadia.org/drepper/tls.pdf
 static inline uintptr_t _mi_thread_id(void) mi_attr_noexcept {
   uintptr_t tid;
   #if defined(__i386__)
   __asm__("movl %%gs:0, %0" : "=r" (tid) : : );  // 32-bit always uses GS
   #elif defined(__MACH__)
-  __asm__("movq %%gs:0, %0" : "=r" (tid) : : );  // x86_64 macOS uses GS
+  #if defined(KONAN_MI_MALLOC)
+  #include <TargetConditionals.h>
+    #if TARGET_OS_EMBEDDED // iOS/tvOS/watchOS devices.
+      tid = pthread_self();
```

Comment:
At least on 64-bit iOS x18 seems to be TLS register (https://stackoverflow.com/questions/28109826/arm64-using-gas-on-ios), but not sure if we rely on that here or not.

Reply:
Thank you